code/modules/surgery/organs/tongue.dm

```diff
@@ -286,21 +286,68 @@
 	modifies_speech = TRUE
 	taste_sensitivity = 32
 
-/obj/item/organ/internal/tongue/zombie/modify_speech(datum/source, list/speech_args)
-	var/list/message_list = splittext(speech_args[SPEECH_MESSAGE], " ")
-	var/maxchanges = max(round(message_list.len / 1.5), 2)
-
-	for(var/i = rand(maxchanges / 2, maxchanges), i > 0, i--)
-		var/insertpos = rand(1, message_list.len - 1)
-		var/inserttext = message_list[insertpos]
+// List of english words that translate to zombie phrases
+GLOBAL_LIST_INIT(english_to_zombie, list())
+
+/obj/item/organ/internal/tongue/zombie/proc/add_word_to_translations(english_word, zombie_word)
+	GLOB.english_to_zombie[english_word] = zombie_word
+	// zombies don't care about grammar (any tense or form is all translated to the same word)
+	GLOB.english_to_zombie[english_word + plural_s(english_word)] = zombie_word
+	GLOB.english_to_zombie[english_word + "ing"] = zombie_word
+	GLOB.english_to_zombie[english_word + "ed"] = zombie_word
+
+/obj/item/organ/internal/tongue/zombie/proc/load_zombie_translations()
+	var/list/zombie_translation = strings("zombie_replacement.json", "zombie")
+	for(var/zombie_word in zombie_translation)
+		// since zombie words are a reverse list, we gotta do this backwards
+		var/list/data = islist(zombie_translation[zombie_word]) ? zombie_translation[zombie_word] : list(zombie_translation[zombie_word])
+		for(var/english_word in data)
+			add_word_to_translations(english_word, zombie_word)
+	GLOB.english_to_zombie = sort_list(GLOB.english_to_zombie) // Alphabetizes the list (for debugging)
 
-		if(!(copytext(inserttext, -3) == "..."))//3 == length("...")
-			message_list[insertpos] = inserttext + "..."
+/obj/item/organ/internal/tongue/zombie/modify_speech(datum/source, list/speech_args)
+	var/mob/living/carbon/human/user = source
 
-		if(prob(20) && message_list.len > 3)
-			message_list.Insert(insertpos, "[pick("BRAINS", "Brains", "Braaaiinnnsss", "BRAAAIIINNSSS")]...")
+	// only high functioning zombies can bypass speech limitations
+	if(iszombie(user) && !is_species(user, /datum/species/zombie/infectious))
```

Comment:
I think all zombies should speak like this, or anyone with this tongue, it's more comedic.

Reply:
Applied with https://github.com/tgstation/tgstation/pull/69899/commits/5288fbc7fcf1350fd62665389b1dabf2442dc2e2